### DISCUSSION TITLE
Station Statistics Page

### DISCUSSION THREAD

**Discussion by JonFear** [Daveiano/weewx-wdc#140] [Ideas]
Provide a page to obtain station statistics like uptime, battery status, link strengths etc. I believe that all these metrics are catered for by Weewx itself.

With batteries becoming more used in installations, this could be a very useful page...

Jon

**Comment by Daveiano** [comment]
Yes, this is indeed a useful addition. As far as I know, this data is sent from the station to WeeWX, sadly my station does not provide this data. 

Many skins include a telemetry page by default, like the seasons skin, for example: https://www.oelsnitz-im-erzgebirge.de/telemetry.html
(Template: https://github.com/weewx/weewx/blob/master/skins/Seasons/sensors.inc)

The observations shown by the seasons skin are the following (from skin.conf):

```
sensor_connections = rxCheckPercent, signal1, signal2, signal3, signal4, signal5, signal6, signal7, signal8
sensor_batteries = outTempBatteryStatus, inTempBatteryStatus, rainBatteryStatus, hailBatteryStatus, snowBatteryStatus, windBatteryStatus, uvBatteryStatus, txBatteryStatus, batteryStatus1, batteryStatus2, batteryStatus3, batteryStatus4, batteryStatus5, batteryStatus6, batteryStatus7, batteryStatus8
sensor_voltages = consBatteryVoltage, heatingVoltage, supplyVoltage, referenceVoltage
``` 

Does your station provide any of these observations? If so, it would be great if you could send me your database, so I have some test data.

I will put this on the list for the next feature release but I have to read into these observations and do some research ;)

**Comment by JonFear** [comment]
Hi Dave

Yes, my system does provide. It is the Ecowitt WS90 using the GW1000 driver so plenty of data there...

https://github.com/gjr80/weewx-gw1000

BW

Jon

**Reply by Daveiano** [reply]
Oh ok, I am using the https://github.com/matthewwall/weewx-interceptor driver with a DNT WeatherScreen PRO.

I am wondering if the GW1000 driver also works for my station, I guess I need to test this out.

Any feedback/ideas/examples for the station's stats page are very appreciated!

**Reply by JonFear** [reply]
I was going to say that the Ecowitt Gateway package may be an improvement for you however looking at the pdf for your system, it looks like everything goes via the console. I am not sure that the Eco/GW package allows for that. All my comms are via the newer GW2000 interface which has greater functionality built into it for the WS90 plus future developments. The GW1000/1100 are now pretty much discontinued.

I guess the only way to tell is to try it and see.... I wonder if you could use a GW2000 with your system. It would then future proof your system going forward.

Jon

**Reply by Daveiano** [reply]
> I guess the only way to tell is to try it and see.... I wonder if you could use a GW2000 with your system. It would then future proof your system going forward.

Yes this sounds like the way to go, I will try with the gw1000 driver first. A system upgrade is planned, but not for this year, first I want to build my allsky cam ;)

**Reply by JonFear** [reply]
All Skycam! Nice, for GMN I assume? 

Whilst there are some nice systems out there, me and a friend proved that simple cameras from Aliexpress could really cut the mustard. The North Camera is just a new one using the IMX335 sensor and a 3.6mm F1.0 lens. This will be replaced with the same camera but with a 6mm F1.0 lens soon. When this happens I will; upgrade the Cloudcam with the new module and a 2.8mm F1.0 lens also coming in soon. Hopefully this should provide sufficient sky coverage. I seem to remember that the current lens is 1.8mm but something like F1.4 which is just not fast enough. 

There are stand alone multi output POE boards out there as well. If you can get a large enough optically clear dome, it would be quite simple just to buy the cameras and feed them into something a bit better than a Raspberry Pi. Something like and HP 400 MSFF with an i7 with plenty of cores running Linux. The actual engineering is not that complex.

The Chinese do some lovely Optically correct glass domes. A group buy would be the easy way of getting something better than the standard acrylic dome used on CCTV cameras...

BTW, thank you for the release. I was going to say perhaps we need to think about a separate variables file which does not get overwritten on re-install. Given the amount of variables I have just introduced for you and everyone else...

Jon

**Reply by JonFear** [reply]
Not sure if this helps but this page is from the Ecowitt Gateway git. It details about the Ecowitt sensor battery and such information.

https://github.com/gjr80/weewx-gw1000/blob/master/supplementary/skins/Seasons/sensors.inc

This is a snip from my weewx.conf file accumulator section which looks like it was added by the GW1000 install...

 [[wh40_batt]]
        extractor = last
    [[wh26_batt]]
        extractor = last
    [[wh25_batt]]
        extractor = last
    [[wh24_batt]]
        extractor = last
    [[wh65_batt]]
        extractor = last
    [[wh31_ch1_batt]]
        extractor = last
    [[wh31_ch2_batt]]
        extractor = last
    [[wh31_ch3_batt]]
        extractor = last

Hope this makes some sense to you...

Jon

**Reply by Daveiano** [reply]
> All Skycam! Nice, for GMN I assume?

Oh no, I was doing some research lately because of your webcam request, and I found this: https://github.com/thomasjacquin/allsky
Looks very promising and I want to try it out.

For the first attempt, I will use a ZWO ASI 120MC I have lying around here in my Astro bag. I have read the requirements something about min RAM of 2GB, and I have a Pi also waiting for work :)

> I was going to say perhaps we need to think about a separate variables file which does not get overwritten on re-install. Given the amount of variables I have just introduced for you and everyone else...

Remember to make changes you want to persist between updates to make in the weewx.conf instead of the skin.conf.

> Not sure if this helps but this page is from the Ecowitt Gateway git. ...
> [...]
> Hope this makes some sense to you...


Yes, this seems to make sense!

I will have a look at the GW1000 driver within the next few weeks, thanks in advance for your help!

**Reply by Daveiano** [reply]
@JonFear Sadly the GW1000 does not work for my weather station. Guess I will need to upgrade it at some time (I first bought the station and then I explored weewx so I did not consider this when buying).

So, you would recommend EcoWitt? Sorry, I am not up to date with the weather stations technics :D

**Comment by JonFear** [comment]
Hi Dave

Sad indeed. Yes, given their price point, I would recommend Ecowitt. Many of the products out there are in fact "badged" versions of their products. 

This link seems to approve of Ecowitt... https://theweatherstationexperts.com/ecowitt-weather-station/ I think they are being a little hard on the Wittboy however, I have never had any troble with it yet, the Rain sensor does need work but that is why you always have a back up, just the same as I have a separate temperature and Humidity sensor. Yes it costs a little more but actually works out cheaper in the long run. My temperature sensor is in a proper Stevenson screen to provide some level of accuracy. 

I have come from the Davis VP2 system which whilst very nice is just do damned expensive to warrant buying it for amateur use. With Davis you pay for everything, even getting access to your own data... 

I have the WS90 Wittboy (WS2001) which is very good but would always retain the traditional WH40 Rain gauge to cross check the rain fall figures. The Haptic rain sensor is still very much a work in progress, they will get it right soon. The ultrasonic wind gauge just works and is impressive. I started with the Gw1103 which was also very good. The WS90 has the newer GW2000 hub unit.

I have taken a quick look at the prices and I am shocked at what I see. Go direct to Ecowitt https://shop.ecowitt.com/products/wittboy?variant=41794898165922 and the WS2001 is US$199 plus tax and shipping. Amazon UK it is £234, Amazon.de is E324. A lot more than I paid under Kickstarter... 

If I were to buy now I would get 1x WS2001 suite which has the GW2000, 1x WH40 Rainfall sensor and 1x WH32 outdoor temperature/humidity sensor. That gives the best coverage for all metrics.

It is interesting to note that Ecowitt are developing a wireless water valve which "should" use the data from their weather stations to control it. I am getting some Wh51 soil temperature sensors ready for the summer. I hope to be able to use this with our irrigation system...

For convenience I would buy from Amazon UK for speed but from Ecowitt direct if I could wait for up to 14 days... Easter holidays are soon, I would want to do it then. I have to update the WS90 firmware that weekend as a matter of fact. I also have to take my antenna mast down for maintenance that weekend.
   
HTH

Jon

**Reply by Daveiano** [reply]
Thank you for your detailed report! Very interesting information. Yeah, I had an eye on Ecowitt for quite some time. I especially like the modular systems- great to have the option to expand the system with a lightning detector or Air quality sensors. 

As I said, I won't make the purchase any time soon, but now I know what my next station may be, which is great!

Back to topic: For the station stats page, it would be great if you could provide me with a snapshot of your weewx database so I have some test data for development.

**Reply by JonFear** [reply]
Hi Dave

I have tarred and GZ'd my DB. Do you have a secure drop box or do you want me to email a link?

Jon

**Reply by Daveiano** [reply]
Sorry, not using dropbox so far. You can just mail it to david.baetge@gmail.com, if that no problem for you?

**Reply by JonFear** [reply]
NP, email sent...

Jon

**Comment by devonian** [comment]
Hi Dave,

If it helps you, I can let you have a copy of my Davis VP2 weewx sqlite database - it's currently ~170Mb but could make it available for you to download as I don't think it would email !
I too would like to see the ISS battery status etc as you would in the Seasons skin.

Nigel

**Reply by Daveiano** [reply]
Hi Nigel, sure that would help! The more data the better, more data for testing!

**Comment by devonian** [comment]
Dave,
Pinged you an email.
Please let me know when done ;-)
Nigel

**Reply by Daveiano** [reply]
Got it and downloaded it! Thank you very much! I will have a look at the data and how to bring this information into the skin for the next release.

**Comment by Daveiano** [comment]
I created #154 from the feedback here. Feel free to comment for more ideas!

By the way: I would call the page just "Station". Is there a better term for this kind of observation like Signal Quality and Batter status?

**Reply by JonFear** [reply]
Seems logical to me...!

**Comment by devonian** [comment]
In weewx skin this kind of data is called "Sensor Status" so follow that for familiarity?

**Reply by JonFear** [reply]
Well I guess so, commonality between projects also makes sense...

**Reply by Daveiano** [reply]
So it shall be "Sensor Status". The new page will be included in the next feature release - 3.2.0, thank you!

**Comment by Daveiano** [comment]
@JonFear @devonian 

Got some time to work on this and here is the outcome so far: 

![Sensor-Status-Sensor-Stauts-page-2-firefox-linux](https://user-images.githubusercontent.com/3184897/235546017-effab129-0f48-4aef-ae5e-fa5a59efcc29.png)

You can hide the data-tables if you want. The timeperiod used in the charts (the example uses a week) is configurable. Which observations are shown is also configurable. But that's how it will look for most people (I believe most stations actually provide `rxCheckPercent` and `consBatteryVoltage` and `txBatteryStatus`).

Any feedback is appreciated!

**Comment by JonFear** [comment]
Hi Dave

At first glance, it looks good. Obviously it will be busier as additional devices are added. Not sure that graphing will be needed for everything but certainly for the main components like the actual station and perhaps things like additional rain gauge and outside thermometers graphing makes sense as it allows for trend identification.

I like it!

All the best

Jon

**Comment by devonian** [comment]
Hi Dave,

Looks fine to me, especially as it will be configurable.
Thumbs up from me!

Nigel

**Comment by JonFear** [comment]
HI Dave

I eventually had a reply regarding my missing data. This is what was said.

> Jon,

No problems. The console output shows the Ecowitt gateway driver is populating loop packets with sensor status data, eg ws90_sig, wh57_batt etc. I assume the sensor data is complete for all sensors registered with your gateway device (I cannot tell this, only you can). WeeWX is then accumulating this data and it appears in WeeWX generated archive records.

Looking at the weewx-wdc GitHub page it appears to use a similar system to the Seasons skins for displaying sensor status data. The default fields to be displayed are set in sensor-status.html.tmpl and these can be overridden in skin.conf. The default appears to include fields such as rxCheckPercent, consBatteryVoltage, inTempBatteryStatus, rainBatteryStatus etc. None of the Ecowitt sensor state fields are included so I expect you need to override/augment the sensor state fields to be displayed in skin.conf. How you do this (ie which settings to adjust) is best discussed with the skin author.

On thing you will need to consider is whether or not to save the sensor state data to archive. It is not necessary to do so if only displaying current values (ie values from the current archive record), but if you wish to display any aggregates based on historical data you will need to do so. Fortunately it is easy to do, just modify your database schema using wee_database to include the desired fields and WeeWX will take care of the rest. Refer to [Modifying an existing database](http://weewx.com/docs/latest/customizing.htm#Modifying_an_existing_database) in the Customization Guide.

Gary

I have yet to take a good look at what Gary has said, been too busy rebuilding the garden... However the sensor page displays nothing. Any thoughts?

Jon

**Comment by Daveiano** [comment]
Hi Jon, I wrote to your original post at https://groups.google.com/g/weewx-user/c/AwhvlJ-DRUQ/m/bzLWZV19DQAJ. Seems to be the right place since Gary is also involved

**Comment by Daveiano** [comment]
Hi Jon, with the [latest release](https://github.com/Daveiano/weewx-wdc/releases/tag/v3.3.0) you should be able to display at least a few observations on the page. For displaying charts, you would need to save those observations to the database like Gary said:

> On thing you will need to consider is whether or not to save the sensor state data to archive. It is not necessary to do so if only displaying current values (ie values from the current archive record), but if you wish to display any aggregates based on historical data you will need to do so. Fortunately it is easy to do, just modify your database schema using wee_database to include the desired fields and WeeWX will take care of the rest. Refer to [Modifying an existing database](http://weewx.com/docs/latest/customizing.htm#Modifying_an_existing_database) in the Customization Guide.

See my comments on google-groups (I also added a list of all observations your station emits):

**Batteries**
> Observations that show something like "Battery OK" or "Battery LOW" (where 0 is OK and 1 is LOW): `wh31_ch1_batt, wh31_ch2_batt, wh32_batt`, and `wh41_ch1_batt, wh57_batt` (see https://github.com/gjr80/weewx-gw1000/wiki/Sensor-battery-states). This kind of observation needs to be added to [DisplayOptions]sensor_battery_status in skin.conf.

**Stat tiles**
You can theoretically add all observations to `[DisplayOptions]sensor_stat_tile_observations`, they should be shown in the stat tiles.

Please let me know if that helps!
